### ISSUE TITLE
Icon for Emulationstation

### ISSUE THREAD

**Issue by dat-atr-ash** [SylEleuth/gruvbox-plus-icon-pack#46]
Hello, dear SylEleuth,

first of all, thank you for this beautiful icon pack!

I needed an icon for Emulationstation, a frontend for Retroarch.

I took the bright template you provide in this repo and adjusted the colors of this file I found in the Emulationstation repo: https://github.com/Aloshi/EmulationStation/blob/master/data/resources/splash.svg.

Note that there are two overlapping 'shadows' and that the resulting colors are not strictly part of the palette.

I don't know if the licenses are compatible (GPL 3 here, MIT there).

If licenses are compatible and you like the result, please feel free to incorporate this icon in your pack. Or let me know if I should open a PR. Never done that, but I'll be happy to learn how to do it.

Also, please let me know if metadata has to be edited before this can be used, should you decide to do so.

![emulationstation](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/46b24bf0-be7a-434d-8fff-be22da4fe3f1)

**Comment by SylEleuth** [comment]
Hi.

Thank You for your kind words. Let's jump straight to icons making stuff.
1. Main part of the above icon are letters "ES", everything else is a background. So colour of the base should be darkest blue, filling the whole space (no need for gradient like in the template, because next shade of the blue is too bright)
2. Base of the original icon has 2 slated corners that have to be dropped here because all corners are round.
3. Main dark colour is #282828 and main bright is @ebdbb2. Anything above that spectrum (#1d2021 and #fbf1c7ff respectively) should be used only in specific situations, e.g. when the main element is bright (#ebdbb2) and You need to put something brighter on it. "ES" letters should be in standard bright, even though the original ones are #ffffff. You also made them gradient and in original they are solid, and because of gradient's direction and position it looks like "S" is brighter then "E".
4. Shadows colour should be left untouched. If palette doesn't have colours You need, opacity is Your friend.
5. Because base is now bigger than the original one, You would have to extend shadows to the edges.
6. I don't know why original icon has 7 one pixel size red elements that are barely/not visible. If it's a mistake? Can't figure it out. 

I've just crated the icon but instead of showing it here I thought that You would like to try again. It's a few minutes of work for me, but if You decide that's too much for You I can push it right now.

PS. Application icons come in pair with "small" 16px versions located in "16" and "symbolic" folders.

**Comment by dat-atr-ash** [comment]
Hello again,

thank you for your answer. 

And also, thank you for the challenge!
It took me some time to understand exactly what had to be changed. And then way too much time to try and figure out how to fix things.
Neither is English my native language nor did I ever learn anything design related. Or how to use Inkscape, for that matter.

Anyway, here is what I came up with.

![emulationstation_2_16](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/e3793c61-4130-4c75-918e-d0eaf3de700b)
![emulationstation_2](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/cdcce3f1-c8c3-40d8-8018-dbfee32b4904)

Since there are definitely flaws and I'm still not 100% sure I understood the instructions correctly, you might be better off using your own.
But still this was a nice challenge and a fun, if overrun, lunch break (plus some.. :smile: ) .
Thanks a bunch!

**Comment by dat-atr-ash** [comment]
PS. I also did these where the shadows go over the bezel. Obviously that's incorrect from a physical object point of view but I think I like them better. :shrug: 

![emulationstation_1_16](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/4ef75380-65ef-446d-8146-3ab503325cde)
![emulationstation_1](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/5f3452e6-2c90-4fb6-8b5d-eabbcdca50d7)

**Comment by SylEleuth** [comment]
Shadow should go over bezel so the second version is almost perfect. Only 2 comments:
- "ES" should be much bigger, original has only few pixels padding, so I would make it 25-30px in this case (on left and right)
- please save it as "Optimised SVG" (https://github.com/SylEleuth/gruvbox-plus-icon-pack/tree/master/templates#saving)

Symbolic icon is a monochrome icon modified from the big one. In this case just "ES" letters without background. Very easy to make:
![emulationstation](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/33354262/a4a226ed-e91d-44c6-ad9e-5d4cf8c09b4a)
Named `emulationstation.svg` in `16` folder and just the same copy in `symbolic` folder with a name changed to `emulationstation-symbolic.svg`.

Fix the big one and You can push it. If You're confused about monochrome version just leave it and I will add it myself. It will require manual formatting anyway at some point, something that Inkscape won't do, but not now (when I will be doing rework of apps/16). You can take what I did or do it Yourself and push all 3 icons. Or just big one. Your choice.

**Comment by dat-atr-ash** [comment]
Okay. Here is the revised version. 

Settled for 25px margin on the sides, as the difference top/bottom <--> left/right seems more intentional that way.

![emulationstation](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/5b774c40-8fda-4e26-b1b8-e1b0eddc4162)

And here is the whole folder structure including the miniatures:

[ES.zip](https://github.com/SylEleuth/gruvbox-plus-icon-pack/files/14924010/ES.zip)

Basically
.
.
|_16:             emulationstation.svg                   Monochrome 14 width
|_Scalable:   emulationstation.svg                   Coloured 256x256
|_Symbolic: emulationstation-symbolic.svg  Monochrome 14 width

If everything is correct I'll happily learn how to push these changes either tomorrow evening or the day after that.

**Comment by dat-atr-ash** [comment]
PS. By 'happily learn how to', I mean read those guides Github provides.
Just to clear that up.

**Comment by SylEleuth** [comment]
16px:
- maximize icons to the edges (on the sides), if You look at the grid it should be nicely aligned
- no nested groups (no groups whatsoever)
- all elements merged
- if You look at the code there shouldn't be any "transformations", it happens when you take grouped elements and resize them

scalable:
- shadows are misaligned with the base, You might be thinking that "it's just a little bit" and "no one will see it" but I try to polish every elements before push and my ocd wouldn't allow me to just leave it as is

**Comment by dat-atr-ash** [comment]
Hey,

pardon, life happened.
Will come back to it.
Regarding your last message, did I again miss some detail in your documentation?

Have to admit that first I didn't read it at all.
But thought I got it after your secound message, where you explicitly reffered to it.

However, maybe Sunday I'll have the time to complete everything.
Sorry for the delay and thanks for your patience.

Kindest regards,

Chris

Am 10. April 2024 09:20:59 UTC schrieb Syl Eleuth ***@***.***>:
>16px:
>- maximize icons to the edges (on the sides), if You look at the grid it should be nicely aligned
>- no nested groups (no groups whatsoever)
>- all elements merged
>- if You look at the code there shouldn't be any "transformations", it happens when you take grouped elements and resize them
>
>scalable:
>- shadows are misaligned with the base, You might be thinking that "it's just a little bit" and "no one will see it" but I try to polish every elements before push and my ocd wouldn't allow me to just leave it as is
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/SylEleuth/gruvbox-plus-icon-pack/issues/46#issuecomment-2047000630
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by SylEleuth** [comment]
No worries, I assumed that amount of details I've provided overwhelmed You. I was going to ask if You want me to do it, but if You still up to it then go ahead.

**Comment by dat-atr-ash** [comment]
Got a day off today and will continue on the icons.

16 now looks looks like this in text editor:

`<svg width="16" height="16" style="isolation:isolate" version="1.1" xmlns="http://www.w3.org/2000/svg">
  <path d="m2.5352 3c-0.29466 0-0.70563 0.17043-0.91406 0.37891l-1.2422 1.2422c-0.20843 0.20848-0.37891 0.61926-0.37891 0.91406v4.9277c0 0.2948 0.17047 0.70772 0.37891 0.91602l1.2422 1.2422c0.20848 0.20824 0.61935 0.37891 0.91406 0.37891h4.9297c0.2949 0 0.36473-0.17247 0.15625-0.38086l-1.2422-1.2402c-0.20843-0.20848-0.61926-0.38086-0.91406-0.38086l-2.9297 0.001953c-0.29466 0-0.53516-0.24231-0.53516-0.53711v-0.92773c0-0.2949 0.24054-0.53516 0.53516-0.53516h2.9297c0.29471 0 0.53516-0.24231 0.53516-0.53711v-0.92578c0-0.2949-0.24045-0.53711-0.53516-0.53711h-2.9297c-0.29466 0-0.53516-0.24035-0.53516-0.53516v-0.92969c0-0.2948 0.24054-0.53516 0.53516-0.53516h2.9297c0.29471 0 0.70558-0.17047 0.91406-0.37891l1.2422-1.2422c0.20848-0.20839 0.13846-0.37891-0.15625-0.37891h-4.9297zm8 0c-0.29485 0-0.70558 0.17043-0.91406 0.37891l-1.2422 1.2422c-0.20848 0.20848-0.37926 0.62121-0.37891 0.91602l0.0019531 0.92773c3.906e-4 0.29485 0.17043 0.70753 0.37891 0.91602l1.2402 1.2402c0.20829 0.20848 0.61935 0.37891 0.91406 0.37891h2.9297 0.53516c-1.18e-4 0 0 0.24026 0 0.53516v0.92969c0 0.2948-0.24026 0.53516-0.53516 0.53516h-4.9297c-0.29485 0-0.36468 0.17066-0.15625 0.37891l1.2422 1.2422c0.20843 0.20834 0.61916 0.37891 0.91406 0.37891h2.9297c0.29485 0 0.70558-0.17066 0.91406-0.37891l1.2422-1.2422c0.20853-0.20829 0.37891-0.61931 0.37891-0.91406v-0.92969c0-0.2949-0.17047-0.70591-0.37891-0.91406l-1.2422-1.2422c-0.20839-0.20843-0.6194-0.379-0.91406-0.37891l-2.9297 0.0019531c-0.29471 9.47e-5 -0.53516-0.24231-0.53516-0.53711v-0.92773c0-0.2948 0.24026-0.53711 0.53516-0.53711h2.9297c0.29485 0 0.70563-0.17043 0.91406-0.37891l1.2422-1.2422c0.20843-0.20848 0.13841-0.37891-0.15625-0.37891h-4.9297z" fill="#ebdbb2" stroke-width="0" style="isolation:isolate"/>
</svg>
`
![emulationstation](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/81ce62c8-caa6-40e8-8c18-14e94eabdc85)


Scalable:
I assume by base you mean the letters, not the background?

**Comment by SylEleuth** [comment]
16 looks great, scalable with no changes. I didn't mean letters nor background. I meant shadows.
Bottom looks like this:
![Screenshot_20240416_101515](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/33354262/7afe4700-6c76-4686-8f42-e125e272f08c)

**Comment by dat-atr-ash** [comment]
Ooooooookay... Sorry, I've bneen a bit slow here. 

That problem is actually inherited from the template [base-bright-plain.svg](https://github.com/SylEleuth/gruvbox-plus-icon-pack/blob/master/templates/base-bright-plain.svg).
The background and bezel are ever so slightly misaligned. X:6.000 vs X:6.001

I set the bezel to X:6.000 et voilà:
![emulationstation](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/55b89fb8-69d8-4f47-a4e5-3d13da0feefe)

**Comment by dat-atr-ash** [comment]
Okay, this was an absolutely enormous pain in the lower back.
Tremendously enormous I will say.

I tried every path function that could possibly result in what I wanted.
And they all did.
In the exact same way. With the exact same flaw. 

Wow!

So in the end I resorted to what in my mind was brute force and a lot less reliably exact: manually manipulating handles and curves. Didn't even know you could do that with curves.
So at least I learned something. Painfully so, but still.

There will definitely be some imperfections but here is the result:
![emulationstation](https://github.com/SylEleuth/gruvbox-plus-icon-pack/assets/78860035/3f4dea45-8b08-456d-a8dc-4b5244929145)

**Comment by dat-atr-ash** [comment]
Monitor with higher resolution that is. Not per se bigger. But that might also help..

**Comment by SylEleuth** [comment]
Closing for now. You can open when You are ready, or let me know if You want me to do that. One day the time will come and will have to do it anyway.

**Comment by SylEleuth** [comment]
https://github.com/SylEleuth/gruvbox-plus-icon-pack/commit/9f10e83683442b2b080433f55d668296b61de0e7
